internal/pkg/template/templates/workloads/partials/cf/nlb.yml

```diff
@@ -10,25 +14,90 @@ PublicNetworkLoadBalancer:
         - Fn::ImportValue:
             !Sub '${AppName}-${EnvName}-PublicSubnets'
     Type: network
+{{- if .NLB.UDPListenerExists }}
+    SecurityGroups:
+      - !GetAtt NetworkLoadBalancerSecurityGroup.GroupId
+NetworkLoadBalancerSecurityGroup:
+  Metadata:
+    'aws:copilot:description': 'A security group for your network load balancer allowing public traffic'
+  Type: AWS::EC2::SecurityGroup
+  Properties:
+    GroupDescription: Allow traffic to the public facing load balancer on listener ports
+    SecurityGroupIngress:
+{{- range $listener := .NLB.Listener }}
+      - CidrIp: 0.0.0.0/0
+        Description: Allow {{- if eq $listener.Protocol "TLS" }} TCP {{- else }} {{ $listener.Protocol }} {{- end}} traffic from anyone on listener port {{ $listener.Port }}
+        FromPort: {{ $listener.Port }}
+        IpProtocol: {{- if eq $listener.Protocol "TLS" }} TCP {{- else }} {{ $listener.Protocol }} {{- end}}
+        ToPort: {{ $listener.Port }}
+{{- end}}
+    VpcId:
+      Fn::ImportValue:
+        !Sub "${AppName}-${EnvName}-VpcId" 
+    Tags:
+      - Key: Name
+        Value: !Sub 'copilot-${AppName}-${EnvName}-${WorkloadName}-nlb'
+EnvironmentSecurityGroupIngressFromNetworkLoadBalancerSecurityGroup:
+  Type: AWS::EC2::SecurityGroupIngress
+  Metadata:
+    'aws:copilot:description': 'Allow ingress from the network load balancer to services in your environment'
+  Properties:
+    GroupId:
+      Fn::ImportValue:
+        !Sub '${AppName}-${EnvName}-EnvironmentSecurityGroup'
+    IpProtocol: -1
+    SourceSecurityGroupId: !GetAtt NetworkLoadBalancerSecurityGroup.GroupId
+{{- else}} # Old NLB security group for non-UDP traffic
```

Comment:
I would do 
```
{{- else}} {{ /* Old NLB security group for non-UDP traffic */ }}
```
so that this comment doesn't actually get rendered into user's template!